deepchem/feat/base_classes.py

```diff
@@ -73,27 +121,58 @@ def _featurize_complex(self, mol_pdb, complex_pdb):
     raise NotImplementedError('Featurizer is not defined.')
 
 
-class Featurizer(object):
-  """
-  Abstract class for calculating a set of features for a molecule.
+class MolecularFeaturizer(Featurizer):
+  """Abstract class for calculating a set of features for a
+  molecule.
+
+  The defining feature of a `MolecularFeaturizer` is that it
+  uses SMILES strings and RDKIT molecule objects to represent
+  small molecules. All other featurizers which are subclasses of
+  this class should plan to process input which comes as smiles
+  strings or RDKIT molecules. 
 
-  Child classes implement the _featurize method for calculating features
-  for a single molecule.
+  Child classes need to implement the _featurize method for
+  calculating features for a single molecule.
+
+  Note
+  ----
+  In general, subclasses of this class will require RDKit to be installed.
   """
 
-  def featurize(self, mols, verbose=True, log_every_n=1000):
-    """
-    Calculate features for molecules.
+  def featurize(self, mols, log_every_n=1000):
+    """Calculate features for molecules.
 
     Parameters
     ----------
-    mols : iterable
-        RDKit Mol objects.
+    mols : RDKit Mol / SMILES string /iterable
+        RDKit Mol, or SMILES string or iterable sequence of RDKit mols/SMILES
+        strings.
+
+    Returns
+    -------
+    A numpy array containing a featurized representation of
+    `datapoints`.
     """
-    mols = list(mols)
+    try:
+      from rdkit import Chem
+      from rdkit.Chem.rdchem import Mol
+    except ModuleNotFoundError:
+      raise ValueError("This class requires RDKit to be installed.")
+    # Special case handling of single molecule
+    if isinstance(mols, str) or isinstance(mols, Mol):
+      mols = [mols]
+    else:
+      # Convert iterables to list
+      mols = list(mols)
     features = []
     for i, mol in enumerate(mols):
+      if i % log_every_n == 0:
+        logger.info("Featurizing datapoint %i" % i)
       if mol is not None:
+        # Process only case of SMILES strings.
+        if isinstance(mol, str):
+          # mol must be a SMILES string so parse
+          mol = Chem.MolFromSmiles(mol)
```

Comment:
`MolFromSmiles ` sometimes throw error (ex: invalid smiles case)
So, we add the try-catch...?

```python

if mol is not None:
  # Process only case of SMILES strings.
  try:
    if isinstance(mol, str):
      # mol must be a SMILES string so parse
      mol = Chem.MolFromSmiles(mol)
    features.append(self._featurize(mol))
  except:
    features.append(np.array([]))
```